libs/server/Resp/Objects/ObjectStoreUtils.cs

```diff
@@ -65,5 +65,28 @@ private bool AbortWithErrorMessage(int count, ReadOnlySpan<byte> errorMessage)
 
             return true;
         }
+
+        /// <summary>
+        /// Tries to parse the input as "LEFT" or "RIGHT" and returns the corresponding OperationDirection.
+        /// If parsing fails, returns OperationDirection.Unknown.
+        /// </summary>
+        /// <param name="input">The input to parse.</param>
+        /// <returns>The parsed OperationDirection, or OperationDirection.Unknown if parsing fails.</returns>
+        public OperationDirection GetOperationDirection(ReadOnlySpan<byte> input)
+        {
+            string inputString = Encoding.UTF8.GetString(input).ToUpperInvariant();
```

Comment:
Do not create strings please. Strings are expensive. Do this:

param1.ReadOnlySpan.SequenceEqual("RIGHT"u8) ? OperationDirection.Right : (param1.ReadOnlySpan.SequenceEqual("LEFT"u8) ? OperationDirection.Left : OperationDirection.Unknown)

Also, no need to pay the cost of a method call, do this directly in the caller.

Reply:
In case we need case insensitive comparison, you can use ``Ascii.EqualsIgnoreCase`` for .NET 8 (and fallback to string.Equals(.., StringComparison.OrdinalIgnoreCase for .NET 7 and below, or do case masking if the string allocation matters)

Reply:
I understand what you mean. Could you please help me know how to distinguish the .NET version in the program?

Reply:
Well, I guess we should just allow capitalized for now. We can later make it case-insensitive if the spec says so. 
However if you go with case insensitive path, you can conditionally compile lines against specific .NET versions with preprocessor directives (``if #NET8_OR_GREATER`` etc.).

Reply:
Thank you! I will try it. 
And Redis supports ignoring case. Look at the last line
![image](https://github.com/microsoft/garnet/assets/61727602/a9cb8f08-8324-4641-89e9-34c49dc9b228)

Reply:
> Do not create strings please. Strings are expensive. Do this:
> 
> param1.ReadOnlySpan.SequenceEqual("RIGHT"u8) ? OperationDirection.Right : (param1.ReadOnlySpan.SequenceEqual("LEFT"u8) ? OperationDirection.Left : OperationDirection.Unknown)
> 
> Also, no need to pay the cost of a method call, do this directly in the caller.

I think it would be better to encapsulate it into a function? Since there are other places where this function will be used. It seems a bit long if I expand the function directly at the call point?

Reply:
You could make static helper with `MethodImpl.AggressiveInlining` attribute, if the static call is that much of a concern.